src/main/java/org/mule/extension/api/metadata/DefaultMetadataKey.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.api.metadata;
+
+import org.mule.api.metadata.MetadataKey;
+
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * The default {@link MetadataKey} implementation
+ */
+final class DefaultMetadataKey implements MetadataKey
+{
+
+    private String id;
+    private String description;
+    private Map<String, String> properties;
+
+    @Override
+    public String getId()
+    {
+        return id;
+    }
+
+    public void setId(String id)
+    {
+        this.id = id;
+    }
+
+    @Override
+    public String getDisplayName()
+    {
+        return description;
+    }
+
+    @Override
+    public Optional<String> getProperty(String propertyId)
+    {
+        return Optional.ofNullable(properties.get(propertyId));
+    }
+
+    @Override
+    public Map<String, String> getProperties()
+    {
+        return properties;
```

Comment:
this is fine, but make sure that the properties map is immutable. Otherwise it's not safe to return it